### PR TITLE
wordpress: Add wordpressPackages

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

This is the continuation of https://github.com/NixOS/nixpkgs/pull/124383 (thanks to @mohe2015 ) for packaging Wordpress plugins and themes. Similar to nodePackages, this approach uses a script ([wp4nix](https://git.helsinki.tools/helsinki-systems/wp4nix)) to generate Nix expressions using a list of package or theme names in json files.

So the workflow would be:

1. Add the package name to the `wordpress-plugins.json` file
```
[
  "add-widget-after-content"
, "antispam-bee"
, "async-javascript"
[...]
```
2. Run `./generate.sh`
3. For example, use the package `pkgs.wordpressPackages.plugins.antispam-bee` in the Wordpress module like this:

```
services.wordpress = {
  sites."blog.${config.networking.domain}" = {
    plugins = with pkgs.wordpressPackages.plugins; [
      anti-spam-bee
      code-syntax-block
      lightbox-with-photoswipe
      wp-gdpr-compliance
    ];
    themes = [ pkgs.wordpressPackages.themes.geist ];
  };
}; 
```

This is also described in the `README.md`.

Todos 

- [x] Update `README.md`
- [x] Test it

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/web-apps/wordpress/packages/wordplress-plugins.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[
```
Comment (mohe2015) [inline pkgs/servers/web-apps/wordpress/packages/wordplress-plugins.json:1]: typo in filename

File: pkgs/servers/web-apps/wordpress/packages/default.nix [hunk 2]
```diff
@@ -0,0 +1,121 @@
+{ lib, newScope }:
```
Comment (mohe2015) [inline pkgs/servers/web-apps/wordpress/packages/default.nix:1]: License header needs to be preserved

**Comment by onny** [comment]
Strange if I run wp4nix from nixpkgs I get following error
```
2022/09/24 20:18:09 A worker just exited, 1 worker(s) remaining
2022/09/24 20:18:09 info https://plugins.svn.wordpress.org/jetpack/tags/11.3.2
2022/09/24 20:18:09 Cannot list https://plugins.svn.wordpress.org/jetpack/tags/11.3.2: exit status 1
<?xml version="1.0" encoding="UTF-8"?>
<info>
svn: E170013: Unable to connect to a repository at URL 'https://plugins.svn.wordpress.org/jetpack/tags/11.3.2'
svn: E120171: Error running context: An error occurred during SSL communication
2022/09/24 20:18:09 buildPkgQueueWorker: jobs queue empty, exiting
2022/09/24 20:18:09 A worker just exited, 0 worker(s) remaining
```
When building wp4nix locally it works fine :(

**Comment by onny** [comment]
@mohe2015 Unfortunately I couldn't manage to generate wordpressPackage sets using the json files generated by wp4nix, still trying to fix this but this is more difficult. Are you familiar with it?

**Comment by onny** [comment]
@dasJ @ajs124 @SlothOfAnarchy Thank you for your work on wp4nix. I would like to integrate Wordpress packaging into NixOS using the approach in this PR.

I would like to use the generated plugins, themes and language packages using your upstream Nix expression https://github.com/NixOS/nixpkgs/pull/173622/files#diff-e90349d673d80a4089a94ec946a314282f197cb3d431436a633ecd203fab0c8a
```
services.wordpress.sites."localhost" = {
  virtualHost.adminAddr = "hello@example.org";
  plugins = [ pkgs.wordpressPackages.plugins.wp-statistics ];
};
```

Unfortunately it currently fails
```
error: value is a function while a set was expected
```

Do you remember what was the right way to get this integrated?

**Comment by onny** [comment]
Thanks to your work @mohe2015 I was able to get it to work :)

**Comment by onny** [comment]
This PR is ready for review :)

**Pull request merged**